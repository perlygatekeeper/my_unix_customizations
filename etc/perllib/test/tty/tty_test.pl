#!/usr/bin/perl
#
# A perl-5 script to test if run from an interactive shell or
# in the background, as in a process started by cron or at.
#
$name = $0; $name =~ s'.*/''; # remove path--like basename
$usage = "usage:\n$name";

use sigtrap 'handler' => \&handle_ttou, 'TTOU';

$outfile="$name.out";
open(OUTFILE,">>$outfile") || die("$name: Cannot append to '$outfile': $!\n");

if (-t) {
  $string ="I predict that this is a tty\n";
  $string.="You must have run me interactively :)\n";
# $stty=`stty -a`;
  if (-t STDOUT) {
    $string.="I enjoy the attention you give me.\n";
  } else {
    $string.="Although you started me from a prompt you then threw\n" .
	     "me in the background like so much discarded stuff.\n";
  }
} else {
  $string ="I predict that this is *NOT* a tty\n";
  $string.="I am lonely as I execute alone here in the BACKGROUND :(\n";
}
#print         $string;
print OUTFILE $string;

$tty=`tty`;
$date=`date`;
$string=$tty . $date . "\n";
#print         $string;
print OUTFILE $string;
close(OUTFILE);

#$outfile="$name.$$";
#open(OUTFILE,">$outfile") || die("$name: Cannot write to '$outfile': $!\n");
## print OUTFILE `env`;
print OUTFILE `stty -a`;
#close(OUTFILE);

exit 0;

sub handle_ttou {
  print OUTFILE "ARG!\n";
}

__END__

#if ($tty !~ /not a tty/i) {
#  $string="You must have run me interactively :)\n";
#  print         $string;
#  print OUTFILE $string;
#} else {
#  $string="I am lonely as I execute alone here in the BACKGROUND :(\n";
#  print         $string;
#  print OUTFILE $string;
#}

# $string=<>;
# print         $string;
# print OUTFILE $string;
