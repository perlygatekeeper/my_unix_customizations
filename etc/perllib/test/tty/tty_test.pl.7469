HOME=/
PATH=/usr/sbin:/usr/bin:/usr/ucb:/etc/init.d:/usr/common/sbin:/usr/common/bin:/usr/global/sbin:/usr/global/bin:/usr/openwin/bin:/usr/lib/netsvc/yp:/usr/ccs/bin:/usr/ccs/lib:/opt/SUNWspro/bin:/opt/SUNWste/bin:/opt/SUNWrtvc/bin:/opt/cap60/bin:/home/staff/parker/bin:/opt/WordPerfect/wpbin:/usr/local/hpnp/bin:/usr/local/samba/bin
LOGNAME=root
HZ=
TZ=US/Eastern
TERM=xterms
SHELL=/usr/bin/tcsh
HOSTTYPE=sun4
VENDOR=sun
OSTYPE=solaris
MACHTYPE=sparc
SHLVL=1
PWD=/usr/global/etc/perllib/test/tty
USER=root
GROUP=other
HOST=chemistry.ohio-state.edu
REMOTEHOST=finally.mps.ohio-state.edu
LD_LIBRARY_PATH=/usr/lib:/usr/ccs/lib/:/usr/openwin/lib:/usr/ucblib:/usr/dt/lib:/usr/local/lib:/usr/common/lib:/usr/global/lib
MANPATH=/usr/common/man:/usr/global/man:/usr/man:/usr/openwin/man:/opt/SUNWspro/man:/opt/SUNWste/man:/opt/hpnp/man:/opt/cap60/man:/opt/SUNWrtvc/man:/usr/common/lib/perl/man
t=/dev/rmt/0
nt=/dev/rmt/0n
uc=/usr/common
ucb=/usr/common/bin
ucsb=/usr/common/sbin
ucs=/usr/common/src
ucm=/usr/common/man
uce=/usr/common/etc
uci=/usr/common/info
ucl=/usr/common/lib
ug=/usr/global
ugb=/usr/global/bin
ugsb=/usr/global/sbin
ugs=/usr/global/src
ugm=/usr/global/man
uge=/usr/global/etc
ugi=/usr/global/info
ugl=/usr/global/lib
ul=/usr/local
ulb=/usr/local/bin
ulsb=/usr/local/sbin
uls=/usr/local/src
ulm=/usr/local/man
ule=/usr/local/etc
uli=/usr/local/info
ull=/usr/local/lib
PAGER=/usr/common/bin/less
EDITOR=/usr/bin/vi
VISUAL=/usr/bin/vi
OS=SunOS
HOSTNAME=chemistry.ohio-state.edu
OS_RELEASE=5.6
OS_VERSION=Generic_105181-06
MACHINE=sun4u
PROCESSOR=sparc
PLATFORM=SUNW,Ultra-1
INFO=SunOS chemistry.ohio-state.edu 5.6 Generic_105181-06 sun4u sparc SUNW,Ultra-1
ETHERNET=hme0
www=/www/pages
cgi=/www/cgi
cat=/www/pages/events/Catalyst
betha=/www/pages/betha
cbetha=/www/cgi/betha
chembal=/www/pages/betha/chembal
org=/www/pages/organic
corg=/www/cgi/organic
mchem=/www/cgi/microchem
tcl=/www/pages/../modules/tcl
mess=/var/adm/messages
WWW_HOME=http://ti/support/compsupp/Manuals/perl/perl.html
acc=/www/cgi/MEMOS/admin
gnu=/usr/common/src/gnu
major=/usr/common/majordomo/lists
jump=/usr/common/jumpstart
notes=/usr/common/sbin/NOTES
patches=/usr/common/PATCHES
def=/usr/global/etc/defaults
OPENWINHOME=/usr/openwin
PRINTER=2105nw
