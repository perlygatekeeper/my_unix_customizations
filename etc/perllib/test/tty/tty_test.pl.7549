EDITOR=/usr/bin/vi
ETHERNET=hme0
GROUP=other
HOME=/
HOST=chemistry.ohio-state.edu
HOSTNAME=chemistry.ohio-state.edu
HOSTTYPE=sun4
HZ=
INFO=SunOS chemistry.ohio-state.edu 5.6 Generic_105181-06 sun4u sparc SUNW,Ultra-1
LD_LIBRARY_PATH=/usr/lib:/usr/ccs/lib/:/usr/openwin/lib:/usr/ucblib:/usr/dt/lib:/usr/local/lib:/usr/common/lib:/usr/global/lib
LOGNAME=root
MACHINE=sun4u
MACHTYPE=sparc
MANPATH=/usr/man:/usr/openwin/man:/usr/dt/man:/usr/local/man:/usr/common/man:/usr/global/man
OPENWINHOME=/usr/openwin
OS=SunOS
OSTYPE=solaris
OS_RELEASE=5.6
OS_VERSION=Generic_105181-06
PAGER=/usr/common/bin/less
PATH=/usr/bin:/usr/ucb:/usr/ccs/bin:/usr/ccs/lib:/usr/local/bin:/usr/common/bin:/usr/global/bin:/opt/SUNWdiag/bin:/opt/SUNWdxlib/bin:/opt/SUNWgt/bin:/opt/SUNWspro/bin:/opt/SUNWste/bin:/opt/SUNWwabi/bin:/usr/local/sbin:/usr/common/sbin:/usr/global/sbin:/usr/local/hpnp/bin:/usr/sbin:/etc/init.d:/usr/common/samba/bin:/opt/cap60/bin:/usr/lib/netsvc/yp
PLATFORM=SUNW,Ultra-1
PRINTER=2105nw
PROCESSOR=sparc
PWD=/usr/global/etc/perllib/test/tty
REMOTEHOST=finally.mps.ohio-state.edu
SHELL=/usr/bin/tcsh
SHLVL=2
TERM=xterms
TZ=US/Eastern
USER=root
VENDOR=sun
VISUAL=/usr/bin/vi
WWW_HOME=http://ti/support/compsupp/Manuals/perl/perl.html
acc=/www/cgi/MEMOS/admin
betha=/www/pages/betha
cat=/www/pages/events/Catalyst
cbetha=/www/cgi/betha
cgi=/www/cgi
chembal=/www/pages/betha/chembal
corg=/www/cgi/organic
def=/usr/global/etc/defaults
gnu=/usr/common/src/gnu
jump=/usr/common/jumpstart
major=/usr/common/majordomo/lists
mchem=/www/cgi/microchem
mess=/var/adm/messages
notes=/usr/common/sbin/NOTES
nt=/dev/rmt/0n
org=/www/pages/organic
patches=/usr/common/PATCHES
t=/dev/rmt/0
tcl=/www/pages/../modules/tcl
uc=/usr/common
ucb=/usr/common/bin
uce=/usr/common/etc
uci=/usr/common/info
ucl=/usr/common/lib
ucm=/usr/common/man
ucs=/usr/common/src
ucsb=/usr/common/sbin
ug=/usr/global
ugb=/usr/global/bin
uge=/usr/global/etc
ugi=/usr/global/info
ugl=/usr/global/lib
ugm=/usr/global/man
ugs=/usr/global/src
ugsb=/usr/global/sbin
ul=/usr/local
ulb=/usr/local/bin
ule=/usr/local/etc
uli=/usr/local/info
ull=/usr/local/lib
ulm=/usr/local/man
uls=/usr/local/src
ulsb=/usr/local/sbin
www=/www/pages
