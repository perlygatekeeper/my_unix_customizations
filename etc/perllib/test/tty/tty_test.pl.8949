speed 9600 baud; 
rows = 57; columns = 80; ypixels = 859; xpixels = 743;
eucw 1:0:0:0, scrw 1:0:0:0
intr = ^c; quit = ^\; erase = ^?; kill = ^u;
eof = ^d; eol = <undef>; eol2 = <undef>; swtch = <undef>;
start = ^q; stop = ^s; susp = ^z; dsusp = ^y;
rprnt = ^r; flush = ^o; werase = ^w; lnext = ^v;
-parenb -parodd cs8 -cstopb -hupcl cread -clocal -loblk -crtscts -crtsxoff -parext 
-ignbrk brkint ignpar -parmrk -inpck -istrip -inlcr -igncr icrnl -iuclc 
ixon -ixany -ixoff imaxbel 
isig icanon -xcase echo echoe echok -echonl -noflsh 
-tostop echoctl -echoprt echoke -defecho -flusho -pendin iexten 
opost -olcuc onlcr -ocrnl -onocr -onlret -ofill -ofdel 
