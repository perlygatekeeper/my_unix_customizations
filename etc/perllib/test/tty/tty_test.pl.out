I predict that this is a tty
You must have run me interactively :)
I enjoy the attention you give me.
/dev/pty0
Thu Dec 17 15:50:40 EST 2015

